apps/Standalone/src/mcp/app/McpStandard.tsx

```diff
@@ -70,66 +86,78 @@ const AwaitingConnection = () => {
   const resourceIdValidation =
     /^\/subscriptions\/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\/resourceGroups\/[a-zA-Z0-9](?:[a-zA-Z0-9-_.]*[a-zA-Z0-9])?\/providers\/[a-zA-Z0-9-_.]+\/[a-zA-Z0-9-_./]+$/;
   const isValidResourceId = hasValidAppId && resourceIdValidation.test(appId);
 
+  const resourceDetails = {
+    subscriptionId: 'f34b22a3-2202-4fb1-b040-1332bd928c84',
+    resourceGroup: 'TestACSRG',
+    location: 'westus',
+  };
   const {
     data: workflowAppData,
     isLoading: isWorkflowLoading,
     error: workflowError,
-  } = useWorkflowApp(isValidResourceId ? appId : '', hostingPlan, isValidResourceId);
+  } = useWorkflowApp(isValidResourceId ? appId : '', hostingPlan, isValidResourceId && !!appId);
 
   const canonicalLocation = useMemo(
     () => WorkflowUtility.convertToCanonicalFormat(workflowAppData?.location ?? 'westus'),
     [workflowAppData]
   );
 
-  const resourceDetails = useMemo(() => {
-    if (!isValidResourceId) {
-      return null;
-    }
-    return new ArmParser(appId);
-  }, [appId, isValidResourceId]);
-
   const isConnected = !!(isValidResourceId && workflowAppData && resourceDetails && !workflowError && !isWorkflowLoading);
 
-  const services = useMemo(() => {
-    const siteResourceId = new ArmParser(appId ?? '').topmostResourceId;
-    const armUrl = 'https://management.azure.com';
-    const baseUrl = `${armUrl}${siteResourceId}/hostruntime/runtime/webhooks/workflow/api/management`;
-    const defaultServiceParams = { baseUrl, httpClient, apiVersion };
-
-    const searchService = new StandardSearchService({
-      ...defaultServiceParams,
-      apiHubServiceDetails: {
-        apiVersion: '2018-07-01-preview',
-        subscriptionId: resourceDetails?.subscriptionId ?? '',
-        location: canonicalLocation,
-      },
-      isDev: false,
-      unsupportedConnectorIds: ['/subscriptions/#subscription#/providers/Microsoft.Web/locations/#location#/managedApis/gmail'],
-    });
-
-    const hostService: IHostService = {
-      fetchAndDisplayContent: (title: string, url: string, type: ContentType) => console.log(title, url, type),
-      openWorkflowParametersBlade: () => console.log('openWorkflowParametersBlade'),
-      openConnectionResource: (connectionId: string) => console.log('openConnectionResource:', connectionId),
-      openMonitorView: (workflowName: string, runName: string) => console.log('openMonitorView:', workflowName, runName),
-    };
-
-    return {
-      searchService,
-      hostService,
-    };
-  }, [appId, canonicalLocation, resourceDetails]);
+  const { data: tenantId } = useCurrentTenantId();
+  const { data: objectId } = useCurrentObjectId();
+
+  const services = useMemo(
+    () => getServices(appId as string, workflowAppData as WorkflowApp, tenantId, objectId, canonicalLocation),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [workflowAppData, tenantId, canonicalLocation, appId]
```

Comment:
can we include objectId here, and remove disabling exhaustive-dep

Reply:
Let me do it in my next PR, i already started making changes to load operation so can include it there.